tracker/progress.go

```diff
@@ -161,9 +162,7 @@ func (pr *Progress) SentEntries(entries int, bytes uint64) {
 			pr.Next += uint64(entries)
 			pr.Inflights.Add(pr.Next-1, bytes)
 		}
-		// If this message overflows the in-flights tracker, or it was already full,
-		// consider this message being a probe, so that the flow is paused.
-		pr.MsgAppFlowPaused = pr.Inflights.Full()
+		pr.MsgAppFlowPaused = true
```

Comment:
Is this commit changing the meaning of `MsgAppFlowPaused`?

Reply:
Yes. It wasn't very well specified before.

Now `MsgAppFlowPaused` means the flow of **empty** "probe" `MsgApp` messages is paused, if it's ongoing. But if the `MsgApp` flow is not throttled, this flag is ignored/no-op (see `sendAppend`), though still can be `true` or `false`.

I think we should unexport this flag, it's internal and changed its meaning a couple of times (last time in https://github.com/etcd-io/etcd/pull/14633). We don't use it in CRDB. @ahrtr Does etcd use it?

Reply:
We do [use](https://github.com/cockroachdb/cockroach/blob/254dbd247fb8ed352a11439063b29f23a0767f28/pkg/kv/kvserver/replica_raft_overload.go#L121-L123) the `IsPaused()` helper though, which uses this field. The dependency is not trivial here. Maybe `IsPaused()` should be kept up-to-date with the decision-making in `sendAppend`, i.e. `IsPaused() == true` iff `sendAppend` does not send anything.

Reply:
> @ahrtr Does etcd use it?

No, etcd doesn't use it.

Reply:
I reworked the PR in such a way that `MsgAppFlowPaused` can now be unexported (left a TODO for that). The behaviour of `IsPaused()` method hasn't change, although I think it should be removed in the future.